static/js/components/forms/SiteContentForm.test.tsx

```diff
@@ -1,239 +1,236 @@
 import { times } from "ramda"
 import React from "react"
-import sinon, { SinonSandbox, SinonStub } from "sinon"
-import { shallow } from "enzyme"
-
-import SiteContentForm from "./SiteContentForm"
-
-import { defaultFormikChildProps, shouldIf } from "../../test_util"
+import { mount } from "enzyme"
+import { act } from "react-dom/test-utils"
+import { Formik } from "formik"
+
+import SiteContentForm, {
+  FormFields,
+  InnerFormProps,
+  FormProps
+} from "./SiteContentForm"
 import {
   makeEditableConfigItem,
   makeWebsiteConfigField,
   makeWebsiteContentDetail,
   makeWebsiteDetail
 } from "../../util/factories/websites"
 import {
-  componentFromWidget,
   contentInitialValues,
-  fieldIsVisible,
-  newInitialValues,
   renameNestedFields
 } from "../../lib/site_content"
 import { useWebsite } from "../../context/Website"
-
-import {
-  EditableConfigItem,
-  Website,
-  WebsiteContent,
-  WebsiteContentModalState,
-  WidgetVariant
-} from "../../types/websites"
-import { FormSchema } from "../../types/forms"
+import { WebsiteContentModalState, WidgetVariant } from "../../types/websites"
 import { createModalState } from "../../types/modal_state"
+import ObjectField from "../widgets/ObjectField"
+
+// ckeditor is not working properly in tests, but we don't need to test it here so just mock it away
+function mocko() {
+  return <div>mock</div>
+}
+
+jest.mock("../widgets/MarkdownEditor", () => ({
+  __esModule: true,
+  default:    mocko
+}))
+
+jest.mock("../widgets/SelectField", () => ({
+  __esModule: true,
+  default:    mocko
+}))
 
-jest.mock("../../lib/site_content")
 jest.mock("./validation")
 jest.mock("../../context/Website")
 
-describe("SiteContentForm", () => {
-  let sandbox: SinonSandbox,
-    onSubmitStub: SinonStub,
-    configItem: EditableConfigItem,
-    content: WebsiteContent,
-    mockValidationSchema: FormSchema,
-    website: Website,
-    setDirtyStub: SinonStub
-
-  beforeEach(() => {
-    sandbox = sinon.createSandbox()
-    content = makeWebsiteContentDetail()
-    content.content_context = times(() => makeWebsiteContentDetail(), 3)
-    configItem = makeEditableConfigItem(content.type)
-    // @ts-ignore
-    renameNestedFields.mockImplementation(() => [])
-    website = makeWebsiteDetail()
-    // @ts-ignore
-    useWebsite.mockReturnValue(website)
-    onSubmitStub = sandbox.stub()
-    setDirtyStub = sandbox.stub()
-  })
+function setupData() {
+  const content = makeWebsiteContentDetail()
+  content.content_context = times(() => makeWebsiteContentDetail(), 3)
+  const configItem = makeEditableConfigItem(content.type)
+  const website = makeWebsiteDetail()
+  // @ts-ignore
+  useWebsite.mockReturnValue(website)
```

Comment:
If you want:

```ts
import * as Website from "../../context/Website"

const { useWebsite: mockUseWebsite } = Website  as jest.Mocked<typeof Website>

mockUseWebsite.mockReturnValue(website)
```

That's the best way I've currently found to deal with mock typings. 

I came across [`jest.mocked`](https://jestjs.io/docs/jest-object#jestmockedtitem-t-deep--false) the other day which also looks useful...but as far as I can tell it doesn't exist on `@types/jest` yet. Which is kinda funny, since it's all about typing.

Reply:
ooo nice!